### DISCUSSION TITLE
Possible contribution of windows batch runfiles macro

### DISCUSSION THREAD

**Discussion by peakschris** [bazelbuild/bazel#22843] [General]
Hi,

For a recent project I have ported https://github.com/bazelbuild/bazel/blob/master/tools/bash/runfiles/runfiles.bash to windows batch, and I thought it would be excellent to get it upstreamed into bazel repo in a similar location to runfiles.bash so it can be consumed as easily as the bash version.

The code started as windows_utils.bzl from aspect-bazel-lib but is now almost all rewritten by porting the runfiles.bash code. But before you pigeonhole me, I used copilot for the heavy lifting #notabatguy ;-)

@fmeum @meteorcloudy would you be willing to accept this as a contribution?

https://github.com/peakschris/rules_multirun/blob/windows/internal/bazel-lib/windows_utils.bzl

**Comment by meteorcloudy** [comment]
@peakschris This looks very interesting. But I prefer to keep the complexity and maintenance cost low in bazel, I think aspect-bazel-lib might be a better home for it?

**Reply by fmeum** [reply]
This is both amazing and terrifying. I agree that this should be maintained in a location with `.bat` expertise and active usage close-by, which the Bazel repo isn't. It would probably also need a lot of test coverage. 

@alexeagle Would you be willing to host this? I can help with reviews.

**Reply by peakschris** [reply]
@meteorcloudy @fmeum thank you, I'll look at updating the existing windows_utils in aspect-bazel-lib. I agree with the amazing and terrifying!!

But I had a question: I came across this post https://groups.google.com/g/bazel-dev/c/X5fa9rKElCg which says that shell runfiles must be part of bazel because under bzlmod there is no way to get a stable identifier. If that is still true, doesn't it also apply to the windows bat language? I would like not to have to inline 300 lines of batch script into everyones launchers.

Is all of the code actually required? Could it be paired down for windows?
